<a href="https://colab.research.google.com/github/Carolinsrainbow/UC_AnalisisdeDatosEnPython/blob/main/Ev1_Resuelta.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Instrucciones

Descargar el archivo de Excel llamado “Datos.xls” ubicado en la sección de "Bienvenida" del curso.  
Leer atentamente las preguntas.
Realizar cálculos, desarrollos y análisis en base al archivo.
Las respuestas que debes seleccionar están asociadas a los códigos o resultados que obtendrás del trabajo con el archivo.

In [22]:
# Incorporamos los paquetes
import pandas as pd
import numpy as np

In [23]:
# Incorporamos las pestañas de excel como datos

#Creamos GRD
GRD_sheet = pd.read_excel('Datos.xls',sheet_name='GRD')

#Creamos Hospitales
Hosp_sheet = pd.read_excel('Datos.xls',sheet_name='Hospitales')

#Creamos Informaciones
Info_sheet = pd.read_excel('Datos.xls',sheet_name='Información Histórica')


In [5]:
GRD_sheet

ID GRD      Nombre GRD  Costo resolución en Clínica Privada [MM$]
0   GRD1       Coronario                                      7.917
1   GRD2        Hepático                                      3.417
2   GRD3    Respiratorio                                      2.915
3   GRD4           Renal                                      6.188
4   GRD5     Neurológico                                      6.782
5   GRD6  Traumatológico                                      4.881
6   GRD7       Esofágico                                      2.629
7   GRD8   Oftalmológico                                      6.720
8   GRD9    Circulatorio                                      5.524
9  GRD10      Intestinal                                      7.916

In [9]:
Hosp_sheet

Nombre Servicio de Salud  \
0   Metropolitano Norte                           ...   
1   Metropolitano Occidente                       ...   
2   Metropolitano Occidente                       ...   
3   Metropolitano Occidente                       ...   
4   Metropolitano Occidente                       ...   
5   Metropolitano Central                         ...   
6   Metropolitano Central                         ...   
7   Metropolitano Central                         ...   
8   Metropolitano Oriente                         ...   
9   Metropolitano Oriente                         ...   
10  Metropolitano Oriente                         ...   
11  Metropolitano Oriente                         ...   
12  Metropolitano Sur                             ...   
13  Metropolitano Sur                             ...   
14  Metropolitano Sur                             ...   
15  Metropolitano Sur                             ...   
16  Metropolitano Sur                             ...   
17  Metropolitano Sur Oriente                     ...   
18  Metropolitano Sur Oriente                     ...   
19  Metropolitano Sur Oriente                     ...   
20  Metropolitano Sur Oriente                     ...   

                               Nombre Establecimiento  \
0                      Complejo Hospitalario San José   
1              Hospital Adalberto Steeger (Talagante)   
2                     Hospital Dr. Félix Bulnes Cerda   
3                           Hospital San Juan de Dios   
4        Instituto Traumatológico Dr. Teodoro Gebauer   
5                 Hospital Clínico San Borja-Arriarán   
6   Hospital de Urgencia Asistencia Pública Dr. Al...   
7                                  Hospital el Carmen   
8                               Hospital Del Salvador   
9                          Hospital Dr. Luis Tisné B.   
10       Instituto de Neurocirugía Dr. Alfonso Asenjo   
11  Instituto Nacional de Enfermedades Respiratori...   
12                       Hospital Barros Luco Trudeau   
13  Hospital de Enfermedades Infecciosas Dr. Lucio...   
14              Hospital Dr. Exequiel González Cortés   
15                                   Hospital El Pino   
16          Hospital Parroquial de San Bernardo         
17         Complejo Hospitalario Dr. Sótero del Río     
18                     Hospital Clínico de la Florida   
19              Hospital Metropolitano ( ex Militar )   
20         Hospital Padre Alberto Hurtado (San Ramón)   

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto  \
0                                 11                                33   
1                                  6                                11   
2                                  3                                 6   
3                                 29                                47   
4                                 12                                 6   
5                                 17                                26   
6                                 24                                20   
7                                  6                                12   
8                                 22                                24   
9                                  8                                16   
10                                12                                11   
11                                15                                29   
12                                31                                43   
13                                11                                 9   
14                                23                                 8   
15                                12                                20   
16                                 6                                10   
17                                20                                35   
18                                 9                                15   
19                                12   

In [10]:
Info_sheet

Número de llegada                              Hospital al que llega  \
0                      1       Instituto Traumatológico Dr. Teodoro Gebauer   
1                      2                Hospital Clínico San Borja-Arriarán   
2                      3                         Hospital Dr. Luis Tisné B.   
3                      4              Hospital Dr. Exequiel González Cortés   
4                      5                       Hospital Barros Luco Trudeau   
...                  ...                                                ...   
65529              65530                Hospital Clínico San Borja-Arriarán   
65530              65531              Hospital Dr. Exequiel González Cortés   
65531              65532                              Hospital Del Salvador   
65532              65533  Hospital de Enfermedades Infecciosas Dr. Lucio...   
65533              65534  Hospital de Urgencia Asistencia Pública Dr. Al...   

         GRD  Día de llegada  Día de salida  Estadia  
0       GRD3               1              4        3  
1       GRD7               1              5        4  
2       GRD7               1              5        4  
3      GRD10               1             11       10  
4       GRD6               1              6        5  
...      ...             ...            ...      ...  
65529   GRD2             424            428        4  
65530   GRD9             424            429        5  
65531  GRD10             424            434       10  
65532   GRD3             424            427        3  
65533   GRD7             424            428        4  

[65534 rows x 6 columns]

In [7]:
#Arreglamos los campos
Hosp_sheet['Nombre Servicio de Salud'] = Hosp_sheet['Nombre Servicio de Salud'].str.strip()

Pregunta 1: ¿Cuál de las siguientes opciones corresponde al código que permite calcular el total de camas críticas para cada uno de los servicios de salud?

In [8]:
Hosp_sheet[['Nombre Servicio de Salud','Total Camas Críticas']].groupby('Nombre Servicio de Salud').sum()

Total Camas Críticas
Nombre Servicio de Salud                       
Metropolitano Central                       105
Metropolitano Norte                          44
Metropolitano Occidente                     120
Metropolitano Oriente                       137
Metropolitano Sur                           173
Metropolitano Sur Oriente                   127

Pregunta 2
¿Cuáles de los siguientes servicios de salud son los tres servicios que tienen más camas críticas?  

Selecciona las tres respuestas que consideres correctas.

In [9]:
camas_criticas = Hosp_sheet[['Nombre Servicio de Salud','Total Camas Críticas']].groupby('Nombre Servicio de Salud').sum()
camas_criticas.sort_values(by=["Total Camas Críticas"], ascending = False).head(3)



Total Camas Críticas
Nombre Servicio de Salud                       
Metropolitano Sur                           173
Metropolitano Oriente                       137
Metropolitano Sur Oriente                   127

Pregunta 3 ¿Cuál de los siguientes códigos permite calcular el costo promedio diario de atención para el GRD4?  


In [10]:
Hosp_sheet["Costo diario GRD4 [MM$]"].mean()

0.4484761904761904

Pregunta 4
¿Cuál de las siguientes alternativas corresponde al promedio diario (redondeado con 0 decimales) de pacientes que han llegado a la red durante los últimos 10 días (desde el día 415 al día 424)?  

In [16]:
# Creamos la tabla auxiliar donde tomamos los datos para sacar el promedio ordenados por el día de llegada (número)
aux = Info_sheet[['Día de llegada','Número de llegada']].groupby('Día de llegada').count()


Número de llegada
Día de llegada                   
415                           149
416                           157
417                           170
418                           167
419                           159
420                           183
421                           159
422                           160
423                           159
424                            82

In [17]:
#Elegimos los últimos 10 datos (días)
aux.tail(10)

Número de llegada
Día de llegada                   
415                           149
416                           157
417                           170
418                           167
419                           159
420                           183
421                           159
422                           160
423                           159
424                            82

In [18]:
#Sacamos el promedio de los últimos 10 datos (días)
aux.tail(10).mean()

Número de llegada    154.5
dtype: float64

In [19]:
#Para redondear a cero
np.round(aux.tail(10).mean(),0)

Número de llegada    154.0
dtype: float64

Pregunta 5 ¿Cuál de las siguientes alternativas indica el promedio del costo diario de resolución de los 3 GRDs más *costosos* en clínica privada?  

In [25]:
#Ordenamos por precio del costo de mayor a menor
GRD3 = GRD_sheet.sort_values(by=['Costo resolución en Clínica Privada [MM$]'], ascending = False)
GRD3


ID GRD      Nombre GRD  Costo resolución en Clínica Privada [MM$]
0   GRD1       Coronario                                      7.917
9  GRD10      Intestinal                                      7.916
4   GRD5     Neurológico                                      6.782
7   GRD8   Oftalmológico                                      6.720
3   GRD4           Renal                                      6.188
8   GRD9    Circulatorio                                      5.524
5   GRD6  Traumatológico                                      4.881
1   GRD2        Hepático                                      3.417
2   GRD3    Respiratorio                                      2.915
6   GRD7       Esofágico                                      2.629

In [26]:
#Tomamos los 3 datos del top y se saca el promedio
GRD3['Costo resolución en Clínica Privada [MM$]'].head(3).mean()

7.538333333333334

Pregunta 6:
¿Cuál de los siguientes códigos permite calcular la cantidad de días de estadía promedio, por GRD, para los pacientes ingresados durante los últimos 10 días registrados en el archivo (y entonces, desde el día 415 al día 424)?  

In [9]:
#La base viene con un error por lo que necesitamos crear una tabla auxiliar
Info_sheet_aux = Info_sheet.drop(columns=['Hospital al que llega'])

#Se incorpora el condicial de los días y se agrupa por GRD para sacar el promedio
Info_sheet_aux.loc[Info_sheet_aux['Día de llegada']>415].groupby('GRD').mean()

Número de llegada  Día de llegada  Día de salida  Estadia
GRD                                                             
GRD1        64901.601399      420.146853     430.146853     10.0
GRD10       64875.592105      419.947368     429.947368     10.0
GRD2        64802.683824      419.566176     423.566176      4.0
GRD3        64796.965986      419.482993     422.482993      3.0
GRD4        64831.367188      419.703125     425.703125      6.0
GRD5        64888.114504      420.099237     424.099237      4.0
GRD6        64814.208333      419.625000     424.625000      5.0
GRD7        64820.518987      419.645570     423.645570      4.0
GRD8        64816.142857      419.646617     423.646617      4.0
GRD9        64816.314516      419.653226     424.653226      5.0

Pregunta 7:
¿Cuál de las siguientes alternativas corresponde a la diferencia entre el menor y el mayor costo de operación anual de una cama crítica en el sistema de salud (considerando todos los servicios de salud)?  

In [8]:
# Valor Máximo
Max = Hosp_sheet['Costo de operación anual de una cama [MM$]'].max()

#Valor Minimo
Min = Hosp_sheet['Costo de operación anual de una cama [MM$]'].min()

#Calculamos la diferencia
Max-Min

29.499999999999993

Pregunta 8: ¿Cuál de las siguientes alternativas indica la cantidad de pacientes que llegaron a la red de salud el día en que llegaron más pacientes?  



In [15]:
pacientes = Info_sheet[['Día de llegada','Número de llegada']].groupby('Día de llegada').count()
pacientes.max()

Número de llegada    183
dtype: int64

Pregunta 9: ¿Cuál de las siguientes alternativas corresponde al código que permite determinar el promedio del total de camas críticas de cada servicio de salud de la red?

In [18]:
Hosp_sheet[['Nombre Servicio de Salud','Total Camas Críticas']].groupby('Nombre Servicio de Salud').mean()

Total Camas Críticas
Nombre Servicio de Salud                                                
Metropolitano Central                                              35.00
Metropolitano Norte                                                44.00
Metropolitano Occidente                                            30.00
Metropolitano Oriente                                              34.25
Metropolitano Sur                                                  34.60
Metropolitano Sur Oriente                                          31.75

Pregunta 10: ¿Cuál de las siguientes alternativas corresponde al código para determinar el costo de instalar las 3 camas nuevas menos costosas en cada servicio de salud?

In [35]:
costo_camas = Hosp_sheet[['Nombre Servicio de Salud', 'Costo de instalar una cama [MM$]']].groupby('Nombre Servicio de Salud').min()
costo3camas = costo_camas[['Costo de instalar una cama [MM$]']]*3
costo3camas

Costo de instalar una cama [MM$]
Nombre Servicio de Salud                                                            
Metropolitano Central                                                           3768
Metropolitano Norte                                                             4221
Metropolitano Occidente                                                         3198
Metropolitano Oriente                                                           3168
Metropolitano Sur                                                               3153
Metropolitano Sur Oriente                                                       3084